File: RepoTransBench/target_projects/Java/C#/JulienGenoud_android-percent-support-lib-sample/public_tests/PercentLinearLayoutPublicTest.cs
using System;
using Xunit;
using PercentLinearLayoutLib;

namespace PercentLinearLayoutLib.PublicTests
{
    public class PercentLinearLayoutPublicTest
    {
        private PercentLinearLayout percentLinearLayout;

        public PercentLinearLayoutPublicTest()
        {
            object mockContext = new object();
            object mockAttrs = new object();
            percentLinearLayout = new PercentLinearLayout(mockContext, mockAttrs);
        }

        [Fact]
        public void TestGenerateLayoutParamsPublic()
        {
            object anotherMockAttrs = new object();
            var @params = percentLinearLayout.generateLayoutParams(anotherMockAttrs);
            Assert.NotNull(@params);
            Assert.IsType<PercentLinearLayout.LayoutParams>(@params);
        }

        [Fact]
        public void TestLayoutParamsConstructorsPublic()
        {
            var baseParams = new LinearLayout.LayoutParams(50, 75);
            var copy1 = new PercentLinearLayout.LayoutParams(baseParams);
            Assert.Equal(50, copy1.Width);
            Assert.Equal(75, copy1.Height);

            var copy2 = new PercentLinearLayout.LayoutParams(200, 125);
            Assert.Equal(200, copy2.Width);
            Assert.Equal(125, copy2.Height);

            var marginParams = new LinearLayout.MarginLayoutParams(8, 14);
            var copy3 = new PercentLinearLayout.LayoutParams(marginParams);
            Assert.Equal(8, copy3.Width);
            Assert.Equal(14, copy3.Height);
        }
    }
}

File: RepoTransBench/target_projects/Java/Go/JulienGenoud_android-percent-support-lib-sample/public_tests/percent_linear_layout_public_test.go
package public_tests

import (
	"testing"

	"github.com/stretchr/testify/assert"
)

// Re-define stubs if needed (needed for public_tests in Go module separation)
type Context struct{}
type AttributeSet struct{}

type LinearLayout struct{}
type LayoutParams struct {
	Width  int
	Height int
}
func NewLinearLayoutLayoutParams(w, h int) *LayoutParams {
	return &LayoutParams{Width: w, Height: h}
}
type MarginLayoutParams struct {
	LayoutParams
}
func NewMarginLayoutParams(w, h int) *MarginLayoutParams {
	return &MarginLayoutParams{
		LayoutParams: LayoutParams{Width: w, Height: h},
	}
}

type PercentLinearLayout struct{}
func NewPercentLinearLayout(ctx *Context, attrs *AttributeSet) *PercentLinearLayout {
	return &PercentLinearLayout{}
}
type PercentLinearLayoutLayoutParams struct {
	MarginLayoutParams
}
func NewPercentLinearLayoutLayoutParamsFromBaseParams(base *LayoutParams) *PercentLinearLayoutLayoutParams {
	if base == nil {
		return nil
	}
	return &PercentLinearLayoutLayoutParams{
		MarginLayoutParams: MarginLayoutParams{LayoutParams: LayoutParams{Width: base.Width, Height: base.Height}},
	}
}
func NewPercentLinearLayoutLayoutParams(w, h int) *PercentLinearLayoutLayoutParams {
	return &PercentLinearLayoutLayoutParams{
		MarginLayoutParams: MarginLayoutParams{LayoutParams: LayoutParams{Width: w, Height: h}},
	}
}
func NewPercentLinearLayoutLayoutParamsFromMarginParams(margin *MarginLayoutParams) *PercentLinearLayoutLayoutParams {
	if margin == nil {
		return nil
	}
	return &PercentLinearLayoutLayoutParams{
		MarginLayoutParams: *margin,
	}
}
func (pll *PercentLinearLayout) GenerateLayoutParams(attrs *AttributeSet) *PercentLinearLayoutLayoutParams {
	return NewPercentLinearLayoutLayoutParams(42, 24)
}

// ===== testGenerateLayoutParamsPublic =====
func setupPercentLinearLayoutPublic() *PercentLinearLayout {
	return NewPercentLinearLayout(&Context{}, &AttributeSet{})
}

func TestGenerateLayoutParamsPublic(t *testing.T) {
	pll := setupPercentLinearLayoutPublic()
	params := pll.GenerateLayoutParams(&AttributeSet{})
	assert.NotNil(t, params, "Expected GenerateLayoutParams to return non-nil")
	if _, ok := interface{}(params).(*PercentLinearLayoutLayoutParams); !ok {
		t.Errorf("Expected params to be of type *PercentLinearLayoutLayoutParams")
	}
}

// ===== testLayoutParamsConstructorsPublic =====
func TestLayoutParamsConstructorsPublic(t *testing.T) {
	// Use values: 50, 75; 200, 125; 8, 14
	baseParams := NewLinearLayoutLayoutParams(50, 75)
	copy1 := NewPercentLinearLayoutLayoutParamsFromBaseParams(baseParams)
	assert.Equal(t, 50, copy1.Width, "Width should match base param (public)")
	assert.Equal(t, 75, copy1.Height, "Height should match base param (public)")

	copy2 := NewPercentLinearLayoutLayoutParams(200, 125)
	assert.Equal(t, 200, copy2.Width, "Width should match direct constructor (public)")
	assert.Equal(t, 125, copy2.Height, "Height should match direct constructor (public)")

	marginParams := NewMarginLayoutParams(8, 14)
	copy3 := NewPercentLinearLayoutLayoutParamsFromMarginParams(marginParams)
	assert.Equal(t, 8, copy3.Width, "Width should match margin params (public)")
	assert.Equal(t, 14, copy3.Height, "Height should match margin params (public)")
}